contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -2672,7 +2672,13 @@ TdsPrintTup(TupleTableSlot *slot, DestReceiver *self)
 		 * NBCROW (0xD2). Count the number of nullable columns and build the
 		 * null bitmap just in case while we are at it.
 		 */
-		nullMapSize = (natts + 7) / 8;
+
+		if (sendRowStat)
+			/* Extra bit for the ROWSTAT column */
+			nullMapSize = (natts + 1 + 7) >> 3;
```

Comment:
How did you test this ?

Reply:
Have added a test in the JUNIT test framework and tested with the code mentioned for the issue as well